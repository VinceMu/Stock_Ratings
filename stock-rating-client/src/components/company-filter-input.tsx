import { SyntheticEvent, useState } from "react"
import { Form } from "semantic-ui-react"

interface CompanyFilterInputProps {
    exchangeSymbols: string[];
    onFilterUpdate: (data: CompanyFilterState) => void;
}

interface CompanyFilterState {
    exchangeSymbol: string;
    overallScoreFrom: number;
    overallScoreTo: number;
}

const initialState = {
    exchangeSymbol: "",
    overallScoreFrom: 0,
    overallScoreTo: 50
}

export function CompanyFilterInput({ exchangeSymbols, onFilterUpdate }: CompanyFilterInputProps) {
    const [filterValues, setFilterValues] = useState(initialState);

    const onExchangeSymbolChange = (e: SyntheticEvent<HTMLElement, Event>, { _, value }: { [key: string]: string }) => {
        setFilterValues({ ...filterValues, exchangeSymbol: value })
    }

    const onOverallScoreFromChange = (e: SyntheticEvent<HTMLElement, Event>, { _, value }: { [key: string]: string }) => {
        setFilterValues({ ...filterValues, overallScoreFrom: parseInt(value) })
    }

    const onOverallScoreToChange = (e: SyntheticEvent<HTMLElement, Event>, { _, value }: { [key: string]: string }) => {
        setFilterValues({ ...filterValues, overallScoreTo: parseInt(value) })
    }

    const onSubmit = () => {
        onFilterUpdate(filterValues);
    }

    return (
        <Form onSubmit={onSubmit}>
            <Form.Group>
                <Form.Select
                    name={"exchangeSymbol"}
                    label={<div style={{ color: "whitesmoke" }}>Exchange symbol filter</div>}
                    options={exchangeSymbols.map((val, idx) => ({ key: idx, text: val, value: val }))}
                    placeholder={exchangeSymbols[0]}
                    onChange={onExchangeSymbolChange}
                />
                <Form.Input
                    label={<div style={{ color: "whitesmoke" }}>Overall score from</div>}
                    name="overallScoreFrom"
                    type="number"
                    placeholder={0}
                    onChange={onOverallScoreFromChange}
                />
                <Form.Input
                    label={<div style={{ color: "whitesmoke" }}>Overall score to</div>}
                    name="overallScoreTo"
                    type="number"
                    placeholder={50}
                    onChange={onOverallScoreToChange}
                />
            </Form.Group>
            <Form.Group>
                <Form.Button content='Filter' color={"blue"} />
            </Form.Group>
        </Form >
    )
}
