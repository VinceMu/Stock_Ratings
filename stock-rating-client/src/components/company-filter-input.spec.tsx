import 'react';
import renderer from 'react-test-renderer';
import { CompanyFilterInput } from './company-filter-input';

const exchangeSymbols = ["All", "ASX", "NYSE"]

test('Company filter input matches snapshot', () => {
  const tree = renderer.create(
    <CompanyFilterInput
      exchangeSymbols={exchangeSymbols}
      onFilterUpdate={() => { }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});