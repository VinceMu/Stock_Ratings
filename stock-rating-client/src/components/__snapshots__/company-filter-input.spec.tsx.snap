// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company filter input matches snapshot 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="fields"
  >
    <div
      className="field"
    >
      <div
        style={
          Object {
            "color": "whitesmoke",
          }
        }
      >
        Exchange symbol filter
      </div>
      <div
        aria-describedby={null}
        aria-expanded={false}
        className="ui selection dropdown"
        name="exchangeSymbol"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          className="divider default text"
          role="alert"
        >
          All
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={true}
            className="selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              All
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              ASX
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              NYSE
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        style={
          Object {
            "color": "whitesmoke",
          }
        }
      >
        Overall score from
      </div>
      <div
        className="ui input"
      >
        <input
          aria-describedby={null}
          name="overallScoreFrom"
          onChange={[Function]}
          placeholder={0}
          type="number"
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        style={
          Object {
            "color": "whitesmoke",
          }
        }
      >
        Overall score to
      </div>
      <div
        className="ui input"
      >
        <input
          aria-describedby={null}
          name="overallScoreTo"
          onChange={[Function]}
          placeholder={50}
          type="number"
        />
      </div>
    </div>
  </div>
  <div
    className="fields"
  >
    <div
      className="field"
    >
      <button
        aria-describedby={null}
        className="ui blue button"
        onClick={[Function]}
      >
        Filter
      </button>
    </div>
  </div>
</form>
`;
